golang/cosmos/daemon/cmd/root.go

```diff
@@ -99,24 +119,30 @@ func initTendermintConfig() *tmcfg.Config {
 // initAppConfig helps to override default appConfig template and configs.
 // return "", nil if no custom configuration is required for the application.
 func initAppConfig() (string, interface{}) {
-	// Allow us to overwrite the SDK's default server config.
 	srvCfg := serverconfig.DefaultConfig()
-	// The SDK's default minimum gas price is set to "" (empty value) inside
-	// app.toml. If left empty by validators, the node will halt on startup.
-	// However, the chain developer can set a default app.toml value for their
-	// validators here.
-	//
-	// In summary:
-	// - if you leave srvCfg.MinGasPrices = "", all validators MUST tweak their
-	//   own app.toml config,
-	// - if you set srvCfg.MinGasPrices non-empty, validators CAN tweak their
-	//   own app.toml to override, or use this default value.
-	//
-	// FIXME: We may want to have Agoric set a min gas price in uist.
-	// For now, we set it to zero so that validators don't have to worry about it.
+
+	// FIXME: We may want a non-zero min gas price.
+	// For now, we set it to zero to reduce friction (the default "" fails
+	// startup, forcing each validator to set their own value).
 	srvCfg.MinGasPrices = "0uist"
 
-	return serverconfig.DefaultConfigTemplate, *srvCfg
+	// CustomAppConfig extends the base config struct.
+	type CustomAppConfig struct {
+		serverconfig.Config `mapstructure:",squash"`
+		Swingset            *swingset.SwingsetConfig `mapstructure:"swingset"`
```

Comment:
Maybe add a comment this `"swingset"` here should match `swingset.ConfigPrefix` ?